src/ripple/app/tx/TransactionEngine.cpp

```diff
@@ -241,8 +241,17 @@ TransactionEngine::applyTransaction (
                 throw std::runtime_error ("Duplicate transaction applied to closed ledger");
             }
 
-            // Charge whatever fee they specified.
-            mLedger->destroyCoins (getNValue (txn.getTransactionFee ()));
+            // Charge whatever fee they specified. We break the encapsulation of
+            // STAmount here and use "special knowledge" - namely that a native
+            // amount is stored fully in the mantissa:
+            auto const fee = txn.getTransactionFee ();
```

Comment:
Oh, I see.  There used to be an `STAmount::getNValue()` which did exactly that.  I have a slight bias toward leaving that interface in place in `STAmount` over breaking encapsulation here.  But only barely.  Your call.